qiskit_algorithms/amplitude_amplifiers/grover.py

```diff
@@ -255,16 +195,13 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
             as ``result.top_measurement``.
 
         Raises:
-            ValueError: If a quantum instance or sampler is not set.
-            AlgorithmError: If a sampler job fails.
+            ValueError: If sampler is not set.
+            AlgorithmError: If sampler job fails.
             TypeError: If ``is_good_state`` is not provided and is required (i.e. when iterations
             is ``None`` or a ``list``)
         """
-        if self._sampler is None and self._quantum_instance is None:
-            raise ValueError("A quantum instance or sampler must be provided.")
-
-        if self._quantum_instance is not None and self._sampler is not None:
-            raise ValueError("Only one of quantum_instance or sampler can be passed, not both!")
+        if self._sampler is None:
+            raise ValueError("A sampler must be provided.")
```

Comment:
Its kinda sad that this is the only item you now need and which defaults to None that default is invalid! Given this is the only type for executions that can be passed now with QI gone we could have the sampler setter set in the reference Sampler from Qiskit if its None. The docs would have to change a little but this is more like elsewhere such as QNNs in in ML where the primitive parameter defaults to None but that builds out a reference sampler instance. We can do this with QI gone since you have to use a Sampler now. I think this would be compatible enough and an improvement over having to raise an error. This leaves the parameter order unchanged so its all compatible.

Same thing in (xx)AE algos.